uefi/src/allocator.rs

```diff
@@ -169,3 +169,38 @@ unsafe impl GlobalAlloc for Allocator {
         }
     }
 }
+
+#[cfg(feature = "unstable")]
+unsafe impl core::alloc::Allocator for Allocator {
+    fn allocate(&self, layout: Layout) -> Result<NonNull<[u8]>, core::alloc::AllocError> {
+        // Stable alternative for Layout::dangling()
+        fn dangling_for_layout(layout: Layout) -> NonNull<u8> {
```

Comment:
Since this is gated by `unstable` anyway, I think we might as well enable `alloc_layout_extra` and use `Layout::dangling` directly.